CMakeLists.txt

```diff
@@ -909,6 +907,7 @@ if(openPMD_USE_INVASIVE_TESTS)
         message(WARNING "Invasive tests that redefine class signatures are "
                         "known to fail on Windows!")
     endif()
+    target_compile_definitions(openPMD PRIVATE openPMD_USE_INVASIVE_TESTS=1)
```

Comment:
This macro should never be shipped. It introduces UB and is only used in some testing.
```suggestion
```

Reply:
The macro is still activated only in those special tests and not shipped (see the `if(openPMD_USE_INVASIVE_TESTS)` above). I need this macro inside `Iteration.cpp` in here: https://github.com/openPMD/openPMD-api/blob/8c0d2a21dd3945523dad5af7cd07b380592cd6d2/src/Iteration.cpp#L542
This is the most reliable way to trigger errors to be thrown.

Are there workflows where users compile openPMD with `openPMD_USE_INVASIVE_TESTS` and still install the resulting compiled library? If yes, are there better ways to do this then? 
Just removing this line breaks the parsing test.

Reply:
ah, I see.
ok, extents its meaning but is ok.